### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by bmuhamood**
In this project, you’ll be building an exclusive clubhouse where your members can write embarrassing posts about non-members. Inside the clubhouse, members can see who the author of a post is but, outside, they can only see the story and wonder who wrote it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
+class PostsController < ApplicationController
+  before_action :user_get
+  before_action :set_post, only: %i[show edit update destroy]
```
Comment (lekegitrepo) [inline app/controllers/posts_controller.rb]: * [x] **_Consider the use of `before_action` to restrict access to the `:new` and `:create` methods ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
+  end
```
Comment (lekegitrepo) [inline app/controllers/posts_controller.rb]: * [x] **_It should initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class UsersController < ApplicationController
+end
```
Comment (lekegitrepo) [inline app/controllers/users_controller.rb]: ##### UsersController

##### `new` action

* [x] **_It should have a `new` action with `User.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_up_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))_**

##### `create` action

* [x] **_It should have a create action. Must use `user_params` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_up_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
### Good job :+1: However, some requirements are missing. Check my comments below.

- **_Remove all the logics from the views_**

### Optional:
- **_A helper method to check if the user is signed_in? can be handy here_**
- **_Flash messages for unsuccessful actions_**

#### _Reviewed by Temitayo Adeleke_

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
### HELLO :wave: GREAT PROGRESS SO FAR :+1:  HOWEVER THERE IS A CHANGE THAT IS REQUIRED :arrow_down: 

- [ ] Your application is creating a user:+1 However there appears to be an error when creating a new post kindly check the error below:arrow_down: 

error post;

- [x] Kindly fix the above error and submit the project for another review

**if you have any question kindly reach me on slack @peterrobert**

### HAPPY CODING :computer:

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (addod19) [reply app/controllers/users_controller.rb]: Hi guys, we missed this section. Users controller must have a new and create action as pointed out by @lekegitrepo. We currently have new method which should be rename as `create` and create a new method like this 
``` 
def new
  @user = User.new
end
```

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi **TEAM**, congratulations on a great job done :+1: However, we need to address the comments below before the project can be approved :smile: 
- Kindly add the new method to the `users controller` as pointed out by the previous TSE

KINDLY SUBMIT ANOTHER CODE REVIEW ONCE THIS IS FIXED :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, congratulations on a great job done :+1: Having fixed all the comments outlined by the previous TSE's :slightly_smiling_face: I hereby approve the project :fireworks: 

Go ahead and merge into development :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**